src/utils.rs

```diff
@@ -1,6 +1,36 @@
 //! Libsodium utility functions
 use ffi;
 
+use std::mem;
+
+/// `malloc()` tries to securely allocate enough memory on the heap to
+/// store `len` bytes. The memory region is bounded on both sides by
+/// guard pages, and, when possible, prevented from being paged out to
+/// swap. It is not guaranteed to be aligned. Memory allocated this
+/// way must have `free()` called when it is no longer in use. Callers
+/// must ensure that a null pointer was not returned before using the
+/// result.
+pub unsafe fn malloc(len: usize) -> *mut u8 {
```

Comment:
Actually these are arguably safe, as long as the caller checks the returned pointer for `is_null()`.